includes/content-gate/class-access-rules.php

```diff
@@ -213,10 +213,34 @@ public static function get_subscription_products_options() {
 		}
 		$products = \wc_get_products(
 			[
-				'type'  => [ 'subscription', 'variable-subscription' ],
+				'type'  => [ 'grouped', 'variable-subscription', 'subscription' ],
 				'limit' => -1,
 			]
 		);
+
+		// Filter out grouped products that don't have any subscription products.
+		$products = array_values(
+			array_filter(
+				$products,
+				function( $product ) {
+					if ( $product->is_type( 'grouped' ) ) {
+						$children = $product->get_children();
+						foreach ( $children as $child ) {
+							$child = \wc_get_product( $child );
+							if ( ! $child ) {
+								continue;
+							}
+							if ( $child->is_type( 'subscription' ) || $child->is_type( 'variable-subscription' ) ) {
+								return true;
+							}
+						}
+						return false;
+					}
+					return true;
+				}
+			)
+		);
+
```

Comment:
This seems unrelated to the PR, and I'm not sure I agree with the change... I'd prefer mapping the rule to subscriptions directly rather than proxied via a group.

Groups are more unstable, as you can rearrange and delete them as needed.

Reply:
Ah, my thinking here was that we might want to be able to map access rules to the primary subscription tier product, so this allows selecting group products with the same conditions as that feature. In that scenario, you think it would be safer to require admins to explicitly select every product that might also be in the primary subscription (grouped) product?

Reply:
7829c9f restores the method for access rules, and creates a new `Content_Gate::get_purchasable_product_options()` method which fetches eligible products via `Subscriptions_Tiers::get_tier_eligible_products()` and formats them as an options array.

Reply:
`get_tier_eligible_products()` returns only variable subscriptions or grouped products. No regular subscriptions. Do you think this will be an issue?